### PR TITLE
Feature to add pre-existing whl files without having to build them

### PR DESCRIPTION
**Pull request by bppont**
*Issue #, if available:*
N/A

*Description of changes:*
Adding feature to sdlf-pipLibrary to bring pre-existing whl files without having to build them. 
You can provide python libraries as wheels for use cases involving Glue Python Shell Jobs, EMR, MWAA, etc. This feature enables you to store pre-existing wheels in S3 and enables easy reference via Systems Manager Parameter Store to import those dependencies.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sdlf-pipLibrary/external_wheels.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from io import BytesIO
+
+import boto3
+import botocore
+
+def add_wheels(name, info, team_name, bucket, client_s3, client_ssm):
+    url = info["url"]
+    file_name = url.rsplit(sep="/", maxsplit=1)[1]
+    print(f"file_name: {file_name}")
+    key = f"common_wheels/{team_name}/{file_name}"
```
Comment (jaidisido) [inline sdlf-pipLibrary/external_wheels.py:23]: Could we just call this: `f"wheels/{team_name}/{file_name}"`

File: sdlf-pipLibrary/external_wheels.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                             Bucket=bucket,
+                             Key=key)
+        print(f"s3://{bucket}/{key} uploaded")
+        sleep(5)
+    except botocore.exceptions.ClientError as ex:
+        raise ex
+
+
+    client_ssm.put_parameter(
+        Name=f"/SDLF/Common/{team_name}/{name}",
```
Comment (jaidisido) [inline sdlf-pipLibrary/external_wheels.py:41]: And this: `f"/SDLF/Wheels/{team_name}/{name}"`

File: sdlf-pipLibrary/build.sh [hunk 3]
```diff
@@ -2,11 +2,19 @@
 
 team_name=$1
 
+function add_external_wheels()
+{
+    echo "external_wheels.json exists"
+    artifacts_bucket=$(aws ssm get-parameter --name /SDLF/S3/ArtifactsBucket | grep -o -E "\"Value\": \"[a-z0-9|-]+\"" | awk -F\: '{print $2}' | sed 's/.$//')
+    python3 ../external_wheels.py ${artifacts_bucket:2} $team_name
+}
```
Comment (jaidisido) [inline sdlf-pipLibrary/build.sh:10]: Not needed because of comment below

**Review by jaidisido — CHANGES_REQUESTED** [review]
Looks great, thank you! Just left a couple of suggestions

**Comment by bppont** [comment]
Thanks! Will make the above changes

File: sdlf-pipLibrary/build.sh (hunk 3 above)
Comment (bppont) [reply sdlf-pipLibrary/build.sh]: Removed

File: sdlf-pipLibrary/build.sh [hunk 4]
```diff
@@ -38,7 +46,16 @@ do
...
         python3 ../external_layers.py ${artifacts_bucket:2} $team_name
+        
+        if [ -f "./external_wheels.json" ]; then
+            add_external_wheels
+        fi
+        
         cd ../
+    elif [ -f "./external_wheels.json" ]; then
+        add_external_wheels
+        cd ../
```
[the remark this answers is not in the document]
Comment (bppont) [reply sdlf-pipLibrary/build.sh]: Thanks for this! Updated in last commit

File: sdlf-pipLibrary/external_wheels.py (hunk 1 above)
Comment (bppont) [reply sdlf-pipLibrary/external_wheels.py]: updated to wheels

File: sdlf-pipLibrary/external_wheels.py (hunk 2 above)
Comment (bppont) [reply sdlf-pipLibrary/external_wheels.py]: updated to wheels

**Review by bppont — COMMENTED** [review]
Updated requested changes in the latest commit

**Pull request merged**